src/iteminfo_armor.cpp

```diff
@@ -0,0 +1,417 @@
+            coverage += _( "Either <info>hand</info>. " );
+        } else if( covers( bodypart_id( "hand_l" ) ) && covers( bodypart_id( "hand_r" ) ) ) {
+            coverage += _( "The <info>hands</info>. " );
+        } else if( covers( bodypart_id( "hand_l" ) ) ) {
+            coverage += _( "The <info>left hand</info>. " );
+        } else if( covers( bodypart_id( "hand_r" ) ) ) {
+            coverage += _( "The <info>right hand</info>. " );
+        }
+
+        if( is_sided() && ( covers( bodypart_id( "leg_l" ) ) || covers( bodypart_id( "leg_r" ) ) ) ) {
+            coverage += _( "Either <info>leg</info>. " );
+        } else if( covers( bodypart_id( "leg_l" ) ) && covers( bodypart_id( "leg_r" ) ) ) {
+            coverage += _( "The <info>legs</info>. " );
+        } else if( covers( bodypart_id( "leg_l" ) ) ) {
+            coverage += _( "The <info>left leg</info>. " );
+        } else if( covers( bodypart_id( "leg_r" ) ) ) {
+            coverage += _( "The <info>right leg</info>. " );
+        }
+
+        if( is_sided() && ( covers( bodypart_id( "foot_l" ) ) || covers( bodypart_id( "foot_r" ) ) ) ) {
+            coverage += _( "Either <info>foot</info>. " );
+        } else if( covers( bodypart_id( "foot_l" ) ) && covers( bodypart_id( "foot_r" ) ) ) {
+            coverage += _( "The <info>feet</info>. " );
+        } else if( covers( bodypart_id( "foot_l" ) ) ) {
+            coverage += _( "The <info>left foot</info>. " );
+        } else if( covers( bodypart_id( "foot_r" ) ) ) {
+            coverage += _( "The <info>right foot</info>. " );
+        }
+
+        if( !covers_anything ) {
+            coverage += _( "<info>Nothing</info>." );
+        }
+
+        info.emplace_back( iteminfo( "ARMOR", coverage ) );
+    }
+
+    if( parts->test( iteminfo_parts::ARMOR_LAYER ) && covers_anything ) {
+        info.emplace_back( iteminfo( "ARMOR", _( "Layer: " ) + which_layer( *this ) ) );
+    }
+
+    if( parts->test( iteminfo_parts::ARMOR_WARMTH ) && covers_anything ) {
+        info.emplace_back( iteminfo( "ARMOR", _( "Warmth: " ), get_warmth() ) );
+    }
+
+    insert_separation_line( info );
+
+    if( parts->test( iteminfo_parts::ARMOR_ENCUMBRANCE ) && covers_anything ) {
+        std::string format;
+        const bool sizing_matters = get_sizing( you ) != sizing::ignore;
+        if( has_flag( flag_FIT ) ) {
+            format = _( " <info>(fits)</info>" );
+        } else if( has_flag( flag_VARSIZE ) && sizing_matters ) {
+            format = _( " <bad>(poor fit)</bad>" );
+        }
+        if( sizing_matters ) {
+            const auto sizing_level = get_sizing( you );
+            const auto sizing_info_str = sizing_info( sizing_level );
+            if( sizing_info_str ) {
+                format = sizing_info_str.value();
+            }
+        }
+
+        if( const islot_armor *t = find_armor_data() ) {
+            if( !t->data.empty() ) {
```

Comment:
No need for extra if
```suggestion
        const islot_armor *t = find_armor_data();
        if( t && !t->data.empty() ) {
```

Upd: now that I'm looking through the code, the original PR added a lot of multi-level ifs that should've been `&&` instead...

Upd2: `t` is a horrible name in such context